docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
+
+### Find Function
+
+#### System Architecture of the Find Function<br/>
+
+![](images/FindCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the find command.<br/>
+* UI: The User Interface of the app<br/>
+* FindCommand: The main logic command of the find function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Find Function<br/>
+
+![](images/FindCommand_umlDiagram.png)
+
+1. `FindCommand` gets the input from the User and gets the timetable from `Timetable`
+1. `FindCommand` then searches through the timetable to find event that match the keyword entered by User
+1. `FindCommand` passes the matching events to `UI`
+1. `UI` prints the matching events
+
+#### Sequence Design of Find Function<br/>
+![](images/FindCommand_SequenceDiagram.png)<br/>
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](images/CompareCommand_ClassDiagram.jpg) <br/>
+The System Architecture given above explains the high-level design of the <b>COMPARE</b> command.<br/>
+
+Listed below are some features we will be highlighting:<br/>
+
+* Ui: The User Interface of the app<br/>
+* CompareCommand: The main logic command of the <b>COMPARE</b> function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* User: User(s) of the program<br/>
+* UserList: An arraylist of User<br/>
+
+#### Design of the Compare Function<br/>
+![](images/CompareCommand_UML_Diagram.jpg)<br/>
+
+1) The <b>COMPARE</b> function gets the inputs from the User and retrieves the User's / Target User's<br/>
+timetables based on a specific day. It will then compare the timetables and return an arraylist
+of common timeslots.<br/>
+2)  It will then pass the arraylist onto Ui class to display the range of common <br/>
+timeslots the User and Target User have on a specific day of the week.<br/>
+
+
+#### Sequence Design of Compare Function<br/>
+![](images/CompareCommand_SequenceDesign.jpg) <br/>
+
+#### Improvements to be made to Version 2.0<br/>
+* Display a range of timings instead of individual hours<br/>
+* Remove edge cases<br/>
+
+E.g.<br/>
+* compare /Alex /Mon<br/>
+* compare /Tammy /Mon<br/>
+
+These cases above would no longer output "0000HR, 0100HR" etc but
+instead "0000HR - 1000HR"
+
+### Login Function
+#### System Architecture of the Login Function<br/>
+![](images/LogInCommand_Architecture_Diagram_v001.png)
+* UI: The User Interface of the app<br/>
+* LogInCommand: The main logic command of the Login function<br/>
+* Parser: The main logic that takes input data and builds the command.
+* User: The Class where username, password, etc are stored
+* UserList: The Class that creates an ArrayList where the User objects will be stored
+
+#### Class Diagram
+![](images/LogInCommand_Class_Diagram_v002.png)
```

Comment:
Perhaps some methods like getters should be omitted from the class diagram to simplify and increase readability of the class diagram? 😄